### PR TITLE
feat(ui): build VLAN DHCP status section

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Built DHCP status section in VLAN details. I've updated the design a little bit so that "High availability" is shown as a boolean instead of only showing when it's true, and also some of the model names are now links.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the VLAN details page of a VLAN with/without DHCP (MAAS, external, relayed) and check that the text matches the legacy version. Note that a lot of edge cases will be hard to test on bolla.

## Fixes

Fixes canonical-web-and-design/app-tribe#668

## Screenshot
Screenshot 2022-02-02 at 18-54-06 Default VLAN details bolla MAAS

### CODE REVIEW

File: ui/src/app/base/components/ControllerLink/ControllerLink.test.tsx [hunk 1]
```diff
@@ -1,18 +1,70 @@
...
+  const store = mockStore(state);
+  render(
+    <Provider store={store}>
+      <MemoryRouter>
+        <ControllerLink systemId="abc123" />
+      </MemoryRouter>
+    </Provider>
+  );
+
+  expect(screen.getByLabelText("Loading controllers")).toBeInTheDocument();
```
Comment (huwshimi) [inline ui/src/app/base/components/ControllerLink/ControllerLink.test.tsx:32]: This tests the label, but not that it's displaying a spinner.

File: ui/src/app/base/components/ControllerLink/ControllerLink.tsx [hunk 2]
```diff
@@ -1,16 +1,38 @@
...
-import type { BaseController } from "app/store/controller/types/base";
+import controllerSelectors from "app/store/controller/selectors";
+import type { Controller, ControllerMeta } from "app/store/controller/types";
+import type { RootState } from "app/store/root/types";
+
+type Props = {
+  systemId?: Controller[ControllerMeta.PK] | null;
+};
+
+const ControllerLink = ({ systemId }: Props): JSX.Element | null => {
```
Comment (huwshimi) [inline ui/src/app/base/components/ControllerLink/ControllerLink.tsx:14]: Should this also fetch the controllers to make sure they exist?

File: ui/src/app/subnets/views/VLANDetails/DHCPStatus/DHCPStatus.tsx [hunk 3]
```diff
@@ -1,5 +1,117 @@
...
+        <Link to={subnetsURLs.vlan.index({ id: vlan.relay_vlan })}>
+          {getFullVLANName(vlan.relay_vlan, vlans, fabrics)}
+        </Link>
+      </span>
+    );
+  }
+  return "Disabled";
+};
+
+const DHCPStatus = ({ id }: Props): JSX.Element | null => {
```
Comment (huwshimi) [inline ui/src/app/subnets/views/VLANDetails/DHCPStatus/DHCPStatus.tsx:44]: Should this also fetch the vlans, fabrics and subnets? I wonder if that would be better to do in this component as someone could easily remove it from the parent component and this would break.

**Review by huwshimi — APPROVED** [review]
A couple of small comments.

File: ui/src/app/base/components/ControllerLink/ControllerLink.tsx (hunk 2 above)
Comment (Caleb-Ellis) [reply ui/src/app/base/components/ControllerLink/ControllerLink.tsx:14]: Yeah, since this exists in base components it should probably just handle everything

File: ui/src/app/subnets/views/VLANDetails/DHCPStatus/DHCPStatus.tsx (hunk 3 above)
Comment (Caleb-Ellis) [reply ui/src/app/subnets/views/VLANDetails/DHCPStatus/DHCPStatus.tsx:44]: It could, but I'm of two minds about including all the `fetch()` actions even if a parent component already does that. Taken to its logical extreme we should be running `fetch()` or `get()` functions for every component that retrieves something from state (e.g. in table cell components), which to me seems cumbersome. And in some cases where we're passing an `id` prop down 4 or 5 levels, are we meant to include `fetch()` at every step? But yeah, it does mean the component can be moved anywhere and "just work". I dunno, what do you think?

Comment (huwshimi) [reply ui/src/app/subnets/views/VLANDetails/DHCPStatus/DHCPStatus.tsx:44]: Yeah I think it's a balancing act. `fetch()` is cheap because it's cached. In this case I was particularly thinking about the fact you added `dispatch(fabricActions.fetch());` in the parent component but its not data is not used by that component so I can imagine someone in the future thinking it's no longer needed and remove it.

Comment (Caleb-Ellis) [reply ui/src/app/subnets/views/VLANDetails/DHCPStatus/DHCPStatus.tsx:44]: That's fair. I'll add them in.
